libraries/AP_Scripting/examples/crosstrack_restore.lua

```diff
@@ -0,0 +1,47 @@
+--[[
+
+   example script to show interrupting a mission and then 
+   reseting the crosstracking to the correct line when
+   returning to the mission after the interruption
+--]]
+
+SCRIPT_NAME = "Crosstrack Restore"
+SCRIPT_NAME_SHORT = "XTrack"
+SCRIPT_VERSION = "4.6.0-001"
+
+MAV_SEVERITY = {EMERGENCY=0, ALERT=1, CRITICAL=2, ERROR=3, WARNING=4, NOTICE=5, INFO=6, DEBUG=7}
+FLIGHT_MODE = {AUTO=10, RTL=11, LOITER=12, GUIDED=15, QHOVER=18, QLOITER=19, QRTL=21}
+
+local last_sw = -1
+local AUX_FN = 300
+
+local previous_location
+
+local function update()
+   local sw_current = rc:get_aux_cached(AUX_FN)
+   if not sw_current then
+      -- treat unset as notch1
+      sw_current = 0
+   end
+   if sw_current ~= last_sw then
+      last_sw = sw_current
+      if sw_current == 0 then
+        vehicle:set_mode(FLIGHT_MODE.AUTO)
+        if previous_location ~= nil then
+            vehicle:set_crosstrack_start(previous_location)
+            gcs:send_text(MAV_SEVERITY.INFO, string.format("%s: Restored previous location", SCRIPT_NAME_SHORT))
+        end
+        gcs:send_text(MAV_SEVERITY.INFO, string.format("%s: Switched to AUTO", SCRIPT_NAME_SHORT))
+      else
+        previous_location = vehicle:get_previous_location()
```

Comment:
If nil you could use the current location.

Reply:
Pretty sure that happens anyway, so I'm not sure why I would need/want to do this here.

Reply:
In any of the modes form `get_previous_location` returns false then `previous_location` will be nil here.

Reply:
Which is why line 31 tests for nil before using it. What would you do?

Reply:
It would just make the cross track from where if you flip the switch if your in a mode that does not support `get_previous_location`. 

I guess otherwise you could only accept the interrupt if in a mode that supports it. Here you always switch to loiter with the switch no matter if you have a valid cross track or not.